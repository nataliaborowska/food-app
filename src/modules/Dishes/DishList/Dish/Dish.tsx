import React from 'react';

interface IPropTypes {
  name: string;
  description: string;
  size: number;
  price: number;
}

export const Dish: React.FC<IPropTypes> = (props) => (
  <tr>
    <td>{props.name}</td>
    <td>{props.description}</td>
    <td>{props.size}</td>
    <td>{props.price}</td>
  </tr>
)