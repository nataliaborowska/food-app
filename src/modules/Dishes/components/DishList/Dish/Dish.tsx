import React from 'react';

import {IDish} from '../../../../../store/reducers/dishesReducer';

export const Dish: React.FC<IDish> = (props) => (
  <tr data-test="dish">
    <td>{props.name}</td>
    <td>{props.description}</td>
    <td>{props.size}</td>
    <td>{props.type}</td>
    <td>{props.price}</td>
  </tr>
)