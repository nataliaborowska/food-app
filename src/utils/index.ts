export const dishTypes = ['vegetarian', 'vegan', 'meat', 'gluten free', 'diary free'];