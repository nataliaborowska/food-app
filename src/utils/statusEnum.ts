export enum statusEnum {
  PENDING= 'PENDING',
  SUCCESS= 'SUCCESS',
  FAIL= 'FAIL',
}