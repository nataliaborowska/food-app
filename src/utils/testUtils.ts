export function findByTestAttribute(wrapper: any, dataTestAttribute: string) {
  return wrapper.find(`[data-test='${dataTestAttribute}']`);
}