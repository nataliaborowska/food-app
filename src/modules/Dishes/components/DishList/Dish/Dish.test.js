import {shallow} from 'enzyme';

import {Dish} from './Dish';
import {findByTestAttribute} from '../../../../../utils/testUtils';

describe('testing dish', () => {
  const dishes = {
    name: 'test',
    description: 'test',
    size: 2,
    price: 2,
    _id: 'test',
  };

  it('renders properly', () => {
    const wrapper = shallow(<Dish dishes={dishes} />);
    const dish = findByTestAttribute(wrapper, 'dish');

    expect(dish.length).toBe(1);
  });
});