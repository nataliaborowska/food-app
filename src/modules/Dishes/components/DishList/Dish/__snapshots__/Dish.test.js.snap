// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing dish renders properly 1`] = `
<tr>
  <td />
  <td />
  <td />
  <td />
</tr>
`;
